nixos/modules/system/boot/systemd.nix

```diff
@@ -891,7 +971,9 @@ in
       in optional
       (type == "oneshot" && (restart == "always" || restart == "on-success"))
       "Service '${name}.service' with 'Type=oneshot' cannot have 'Restart=always' or 'Restart=on-success'")
-      cfg.services);
+      cfg.services) ++ optional (cfg.tmpfiles.rules != [ ]) ''
+        systemd.tmpfiles.`rules` is deprecated, please use systemd.tmpfiles.`paths`.
+      '';
```

Comment:
After the design of this new option will be reviewed and approved, perhaps we should also sprint over all the services that use `rules`, and switch them to `paths`?

Reply:
Yes, just wanted to keep this pr small for the beginning, so it's easier to review.